### PR TITLE
Fix usage of search_after in scroll for results

### PR DESCRIPTION
**Pull request by DeLaGuardo**
Fixes potential endless loop in the pagination process. There are two options to do pagination with ductile and ES queries: using offset returned by the previous result or using search_after. Those are entirely different ways to do pagination, and they should not be mixed into any pagination logic.

<a name="qa">[§](#qa)</a> QA
============================

[all_bundles.zip](https://github.com/threatgrid/ctia/files/9555933/all_bundles.zip)

1. Import bundle-0.json

```
curl 'https://visibility.test.iroh.site/iroh/private-intel/bundle/import' \
  -X POST \
  -H 'authorization: Bearer ...' \
  -H 'Content-Type: application/json' \
  --data @bundle-0.json \
  | jq '.results[].id'
```

Each id should look similar to this example: `"https://private.intel.test.iroh.site:443/ctia/sighting/sighting-b7213c75-a43e-4449-aa22-d1a9ea6f78bb"`. Take the last path segment and use it for the next step - `sighting-b7213c75-a43e-4449-aa22-d1a9ea6f78bb`.

2. Check if those sightings can be found in Elasticsearch:

Replace items for values with the list of ids from the previous step

```
curl 'https://es-private-storage.test.iroh.site/v1.1.0_private_ctia_sighting/_search' \
  -X GET \
  -H 'Content-Type: application/json; charset=UTF-8' \
  --data '{
    "sort": {"id": "desc"},
    "stored_fields": ["id"],
    "query": {
      "ids": {
        "values": [
          "sighting-4bdfabdd-0b81-4e9f-acac-446c176b4ba6",
          "sighting-36b3d721-b493-4678-969a-659e662b18fe",
          "sighting-bd451b1e-72e3-44cb-aa26-9fa58dc352ae"
        ]
      }
    }
  }' | jq '.hits.hits[]._id'
```

the result is the list of ids for the next step. The order is important.

3. Please replace all occurrences of a string `[[i]]` in every other bundle with an id from the previous step where `i` is an index in the array:

```
[
  "sighting-4bdfabdd-0b81-4e9f-acac-446c176b4ba6",
  "sighting-36b3d721-b493-4678-969a-659e662b18fe",
  "sighting-bd451b1e-72e3-44cb-aa26-9fa58dc352ae"
]

=>

[
  "https://private.intel.test.iroh.site:443/ctia/sighting/sighting-4bdfabdd-0b81-4e9f-acac-446c176b4ba6", 
  "https://private.intel.test.iroh.site:443/ctia/sighting/sighting-36b3d721-b493-4678-969a-659e662b18fe",
  "https://private.intel.test.iroh.site:443/ctia/sighting/sighting-bd451b1e-72e3-44cb-aa26-9fa58dc352ae"
]
```

4. Import all of them
5. Call bundle export endpoint using ids from step 1.

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: src/ctia/bundle/core.clj [hunk 1]
```diff
@@ -453,8 +454,9 @@
                                                   filters-map
                                                   identity-map
                                                   {:limit limit
-                                                   :sort [{(name related-to) "desc"}
-                                                          {"timestamp" "desc"}]}))))
+                                                   :sort [(name related-to)
+                                                          {"timestamp" "desc"}
+                                                          "id"]}))))
```
Comment (ereteog) [inline src/ctia/bundle/core.clj:459]: if you use `id` I assume that you do not need `timestamp`.

**Review by ereteog — APPROVED** [review]
LGTM

File: src/ctia/bundle/core.clj (hunk 1 above)
Comment (DeLaGuardo) [reply src/ctia/bundle/core.clj:459]: I still need timestamp to sort entities from newest to oldest

Comment (ereteog) [reply src/ctia/bundle/core.clj:459]: Why do we need to sort from newest to oldest?

Comment (DeLaGuardo) [reply src/ctia/bundle/core.clj:459]: that was the original logic that got carried from here - https://github.com/threatgrid/ctia/pull/1285/files#diff-f53321c0a5e441e3481b32d1d20af1f87c9232583d527b9fe9dc468d4d9244b4L405-L413

**Comment by NeelSaini** [comment]
Cherry Picked on TEST in rel v1.105

**Comment by skriths** [comment]
As discussed with @DeLaGuardo, this PR is reverted by follow up and hence, nothing to validate for the same.

**Pull request merged**